src/rules/relayContainerReturnsConnection.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  ASTVisitor,
+  NamedTypeNode,
+  ObjectTypeDefinitionNode,
+  TypeDefinitionNode,
+  ValidationContext,
+} from 'graphql';
+
+import { ValidationError } from '../utils';
+
+export function RelayContainerReturnsConnection(
+  context: ValidationContext,
+): ASTVisitor {
+  const ensureNameDoesNotEndWithContainer = (node: TypeDefinitionNode) => {
+    if (node.name.value.match(/ConnectionContainer$/)) {
+      context.reportError(
+        new ValidationError(
+          'relay-connection-types-spec-pinterest-customization',
+          `Types that end in \`Connection\` must be an object type as per the relay spec. \`${node.name.value}\` is not an object type.`,
```

Comment:
Should this be Types that end in `ConnectionContainer` instead of `Connection`?

Reply:
The idea with this part is to avoid that we have other than ObjectTypeDefinition ending with "Connection". So, with this, we validate that ScalarTypeDefinition, InterfaceTypeDefinition, UnionTypeDefinition, EnumTypeDefinition and InputObjectTypeDefinition doesn't end with Connection.

Validations related to Containers aren't validated by this rule, as this rule verifies if it adheres to [section 2 (Connection Types)](https://facebook.github.io/relay/graphql/connections.htm#sec-Connection-Types) and Containers aren't part of this spec.